Denys Bykov/Task1/StringList.cpp

```diff
@@ -0,0 +1,133 @@
+#include "StringList.hpp"
+
+void StringListInit(StringList* list)
+{
+    *list = static_cast<char**>(malloc(sizeof(char**)));
+    **list = nullptr;
+}
+
+void StringListDestroy(StringList* list)
+{
+    char** pos = *list;
+    for (; *pos; pos++)
+        free(*pos);
+    free(*list);
+    *list = nullptr;
+}
+
+int  StringListSize(StringList list)
+{
+    char** pos = list;
+    for (; *pos; pos++) {}
+    return static_cast<int>(pos - list);
+}
+
+void StringListAdd(StringList* list, char* str)
```

Comment:
Will your code work ok if call this function StringListAdd(nullptr, nullptr); ? The same for almost all functions.